src/contentmanager.cpp

```diff
@@ -942,14 +943,32 @@ bool ContentManager::handleZimFileInMonitoredDirLogged(QString dir, QString file
     return status == MonitoredZimFileInfo::ADDED_TO_THE_LIBRARY;
 }
 
+void ContentManager::MonitoredZimFileInfo::updateStatus(const MonitoredZimFileInfo& prevInfo)
+{
+    Q_ASSERT(prevInfo.status != ADDED_TO_THE_LIBRARY);
+
+    if ( this->lastModified == prevInfo.lastModified ) {
+        this->status = UNCHANGED_KNOWN_BAD_ZIM_FILE;
+    } else {
+        this->status = PROCESS_NOW;
+    }
+}
+
 ContentManager::MonitoredZimFileInfo ContentManager::getMonitoredZimFileInfo(QString dir, QString fileName) const
 {
-    Q_UNUSED(dir);
-    Q_UNUSED(fileName);
-    // TODO: implement properly
-    MonitoredZimFileInfo zfi;
-    zfi.status = MonitoredZimFileInfo::PROCESS_NOW;
-    return zfi;
+    const auto bookPath = QDir::toNativeSeparators(dir + "/" + fileName);
+
+    MonitoredZimFileInfo zimFileInfo;
+
+    zimFileInfo.lastModified = QFileInfo(bookPath).lastModified();
```

Comment:
We may use appropriate constructor here.
I don't like to see structure which may be not fully initialized.

Reply:
In a sense, `MonitoredZimFileInfo::lastModified` is an optional field. It is not used in the other location (in `ContentManager::setMonitoredDirectories()` where a `MonitoredZimFileInfo` object is constructed. IMHO, introducing a constructor for just this incarnation of `MonitoredZimFileInfo` is not justified (besides, an explicit assignment to `.lastModified` make the code more readable).